aws/resource_aws_vpc_dhcp_options.go

```diff
@@ -272,7 +272,7 @@ func resourceDHCPOptionsStateRefreshFunc(conn *ec2.EC2, id string) resource.Stat
 
 		resp, err := conn.DescribeDhcpOptions(DescribeDhcpOpts)
 		if err != nil {
-			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidDhcpOptionsID.NotFound" {
+			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidDhcpOptionID.NotFound" {
```

Comment:
Consider using [`isAWSErr`](https://github.com/terraform-providers/terraform-provider-aws/blob/6512d8fbbdd394398b49ffc9b516dda16f8bc9a3/aws/awserr.go#L11):
```
if isAWSErr(err, "InvalidDhcpOptionID.NotFound", "") {
```
or even extracting to own function
```
func isNoSuchDhcpOptionIDErr(err error) bool {
    return isAWSErr(err, "InvalidDhcpOptionID.NotFound", "")
}
```

Reply:
Maybe check for both forms if you extract to a new function?
It seems both error messages have exactly the same [description](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html).